crates/chain/src/local_chain.rs

```diff
@@ -1,14 +1,116 @@
+//! The [`LocalChain`] is a local implementation of [`ChainOracle`].
+
 use core::convert::Infallible;
 
-use alloc::collections::{BTreeMap, BTreeSet};
+use crate::collections::BTreeMap;
+use crate::{BlockId, ChainOracle};
+use alloc::{sync::Arc, vec::Vec};
 use bitcoin::BlockHash;
 
-use crate::{BlockId, ChainOracle};
+/// A structure that represents changes to [`LocalChain`].
+pub type ChangeSet = BTreeMap<u32, Option<BlockHash>>;
+
+/// Represents a block of [`LocalChain`].
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct CheckPoint {
```

Comment:
This type should be private in my mind. It's an implementation detail for how you can produce an iterator for `BlockId` with a single `Arc::clone`.

Reply:
What are the downsides of exposing `Arc`?

Reply:
`Arc` is now hidden.